### PR TITLE
Create T012_M3_text_ui

### PR DESCRIPTION
**Pull request by Jackinator27**
#10

### CODE REVIEW

File: T012_M3_text_ui [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+        Please type your command (Options: L, S, H, W, B, Q): B
+        Please enter the attribute you want to calculate the best value of the attribute for in terms of grades (Options: Age, StudyTime, Failures, Health, Absences): Age
+        The best value for the attribute Age is 16.89 and the average grade they received is 11.15
+
+    The interface function displays an interface for the user to navigate through using the shell and given prompts. This interface allows the user to load data, sort data, display a histogram, show the worst-for-grades of an attribute, the best-for-grades for an attribute, and finally to quit the program. This interface is a combination of all the previous lab's functions.
+
+    The interface function takes no input parameters.
+
+    Preconditions: The interface function calls many of our previous lab's files, so all files submitted alongside this one must be in the same folder for the function to run. The initial command asking if you want to load data and so forth allows uppercase or lowercase letters, but the sections where the function prompts you to enter in an attribute is case sensitive.
+    """
```
Comment (InochiFumetsu) [inline T012_M3_text_ui:57]: `Filename entered under load data must exist.
"""`

File: T012_M3_text_ui [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+                    loaded_data, cmd3))  # Store the sorted data
+                cmd4 = (input("sorted_data loaded, and sorted by " + str(cmd3) +
+                              ". Do you want to display the data? (Options: Yes or No): ")).capitalize()  # Ask the user if they want to display the data
+                # If user enters invalid command, prompt to retry
+                while (cmd4 not in supported_cmds):
+                    cmd4 = (input(
+                        "Invalid command. Do you want to display the data? (Options: Yes or No): ")).capitalize()
+                if cmd4 == "Yes":
+                    # If the user wants to display the data, print the sorted data.
+                    print(str(sorted_data) + "\n")
```
Comment (InochiFumetsu) [inline T012_M3_text_ui:142]:
```suggestion
                    for i in iter(sorted_data):
                        print(i, ":", sep="")
                        for j in range(len(sorted_data[i])):
                            print("\t", sorted_data[i][j], sep="")
```

File: T012_M3_text_ui (hunk 1 above)
Comment (InochiFumetsu) [inline T012_M3_text_ui:57]:
```suggestion
    Filename entered in load data function of menu must exist.
    """
```

**Review by InochiFumetsu — DISMISSED** [review]

File: T012_M3_text_ui [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+            else:
+                atb2 = input(
+                    "Please enter the attribute you want to calculate the best value of the attribute for in terms of grades (Options: Age, StudyTime, Failures, Health, Absences): ")  # Get the attribute the user wants to calculate the best-for-grades by
+                # Calculate the best-for-grades by the attribute used the loaded_data dictionary
+                atb_val2 = (T012_M3_optimization.maximum(loaded_data, atb2))
+                print("The best value for the attribute " +
+                      str(atb2) + " is " + str(atb_val2[0]) + " and the average grade they received is " + str(atb_val2[1]) + "\n")  # Print the best-for-grades values
+
+        else:
+            print()
```
Comment (InochiFumetsu) [inline T012_M3_text_ui:182]:
```suggestion
```

File: T012_M3_text_ui [hunk 4]
```diff
@@ -0,0 +1,189 @@
...
+                print("File not loaded. Please, load a file first.\n")
+            else:
+                atb2 = input(
+                    "Please enter the attribute you want to calculate the best value of the attribute for in terms of grades (Options: Age, StudyTime, Failures, Health, Absences): ")  # Get the attribute the user wants to calculate the best-for-grades by
+                # Calculate the best-for-grades by the attribute used the loaded_data dictionary
+                atb_val2 = (T012_M3_optimization.maximum(loaded_data, atb2))
+                print("The best value for the attribute " +
+                      str(atb2) + " is " + str(atb_val2[0]) + " and the average grade they received is " + str(atb_val2[1]) + "\n")  # Print the best-for-grades values
+
+        else:
```
[the remark this answers is not in the document]
Comment (InochiFumetsu) [reply T012_M3_text_ui:181]: p. sure the else statement will literally never execute :P

File: T012_M3_text_ui (hunk 3 above)
Comment (InochiFumetsu) [reply T012_M3_text_ui:182]: no need for this if the else won't ever execute.

File: T012_M3_text_ui (hunk 2 above)
Comment (InochiFumetsu) [reply T012_M3_text_ui:142]: I didn't read the descriptions for the functions you and Esteban had to write .. so idk if you're allowed to this or you had to do it the way you had it.. throw it in your code and see if you like?

**Review by Hey1121 — DISMISSED** [review]

**Review by InochiFumetsu — APPROVED** [review]

**Review by MilanxDjordjevic — APPROVED** [review]

**Review by Hey1121 — APPROVED** [review]

**Pull request merged**